timeutil/schedule.go

```diff
@@ -0,0 +1,227 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package timeutil
+
+import (
+	"fmt"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var validTime = regexp.MustCompile(`^([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$`)
+
+type TimeOfDay struct {
+	Hour   int
+	Minute int
+}
+
+// ParseTime parses a string that contains hour:minute and returns
+// an TimeOfDay type or an error
+func ParseTime(s string) (t TimeOfDay, err error) {
+	m := validTime.FindStringSubmatch(s)
```

Comment:
Yes, this will work, I am happy to change it, I am not overly happy with the error return from time.Parse, e.g. ` "parsing time \"xx\" as \"15:04\": cannot parse \"xx\" as \"15\""`. But using time.Parse() is certainly less code.